src/mixins/audit.mixin.ts

```diff
@@ -296,8 +317,101 @@ export function AuditRepositoryMixin<
         delete extras.actionKey;
         const auditLog = new AuditLog({
           actedAt: new Date(),
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          actor: (user?.id as any).toString() ?? '0', //NOSONAR
+
+          actor:
+            options?.actorId ??
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            (user[this.actorIdKey ?? 'id'] as any)?.toString() ?? //NOSONAR
+            '0',
+          action: Action.DELETE_ONE,
+          before: before.toJSON(),
+          entityId: before.getId(),
+          actedOn: this.entityClass.modelName,
+          actionKey: opts.actionKey,
+          ...extras,
+        });
+
+        auditRepo.create(auditLog).catch(() => {
+          //sonarignore:start
+          console.error(
+            `Audit failed for data => ${JSON.stringify(auditLog.toJSON())}`,
+          );
+          //sonarignore:end
+        });
+      }
+    }
+    async deleteAllHard(
+      where?: Where<M>,
+      options?: AuditOptions,
+    ): Promise<Count> {
+      if (!super.deleteAllHard) {
+        return Promise.resolve({count: 0});
+      }
+      if (options?.noAudit) {
+        return super.deleteAllHard(where, options);
+      }
+      const toDelete = await this.find({where}, options);
+      const beforeMap = keyBy(toDelete, d => d.getId());
+      const deletedCount = await super.deleteAllHard(where, options);
+
+      if (this.getCurrentUser) {
+        const user = await this.getCurrentUser();
+        const auditRepo = await this.getAuditLogRepository();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const extras: any = Object.assign({}, opts); //NOSONAR
+        delete extras.actionKey;
+        const audits = toDelete.map(
+          data =>
+            new AuditLog({
+              actedAt: new Date(),
+              actor:
+                options?.actorId ??
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                (user[this.actorIdKey ?? 'id'] as any)?.toString() ?? //NOSONAR
+                '0',
+              action: Action.DELETE_MANY,
+              before: (beforeMap[data.getId()] as Entity).toJSON(),
+              entityId: data.getId(),
+              actedOn: this.entityClass.modelName,
+              actionKey: opts.actionKey,
+              ...extras,
+            }),
+        );
+        auditRepo.createAll(audits).catch(() => {
+          const auditsJson = audits.map(a => a.toJSON());
+          //sonarignore:start
+          console.error(
+            `Audit failed for data => ${JSON.stringify(auditsJson)}`,
+          );
+          //sonarignore:end
+        });
+      }
+      return deletedCount;
+    }
+
+    async deleteByIdHard(id: ID, options?: AuditOptions): Promise<void> {
+      if (!super.deleteByIdHard) {
```

Comment:
Should throw an error if base class does not implement this method? and Ideally the type should not expose this function at all if the baseClass does not have it